Rpackage/R/mmif.R

```diff
@@ -0,0 +1,95 @@
+#' Random Forest of Max Margin Interval Tree
+#'
+#' Performing random forest on ensemble of Max Margin Interval Tree.
+#' 
+#' @param target.mat The response variable of the model
+#' @param feature.mat a data frame containing the feature variables in the model.
+#' @param margin margin paramaters 
+#' @param loss The type of loss; (\code{"hinge"}, \code{"square"})
+#' @param max_depth The maximum depth criteia
+#' @param min_sample The minimum number of sample required 
+#' @param n_trees The Number of trees
+#' @param n_features The number of features to be sampled.
+#' @param n_cpu The number of cores to register for parallel programing of the code, default value is 1 and n_cpu = -1 to select all cores.
+#' 
+#' @return List of ensemble of trees.
+#' 
+#' @author Toby Dylan Hocking, Alexandre Drouin, Torsten Hothorn, Parismita Das
+#' 
+#' @examples
+#' library(mmit)
+#' target.mat <- rbind(
+#'   c(0,1), c(0,1), c(0,1),
+#'   c(2,3), c(2,3), c(2,3))
+#' 
+#' feature.mat <- rbind(
+#'   c(1,0,0), c(1,1,0), c(1,2,0),
+#'   c(1,3,0), c(1,4,0), c(1,5,0))
+#' 
+#' colnames(feature.mat) <- c("a", "b", "c")
+#' feature.mat <- data.frame(feature.mat)
+#' 
+#' trees <- mmif(target.mat, feature.mat, margin = 2.0, n_cpu = -1)
+#' 
+#' @export
+mmif <- structure(function(target.mat, feature.mat, 
+                           max_depth = Inf, margin=0.0, loss="hinge",
+                           min_sample = 1, n_trees = 10,
+                           n_features = as.integer(ncol(feature.mat)**0.5), n_cpu = 1){
+ 
+ 
+  ### parallelize using foreach, see all permutation combination of param grid values
+  ### register parallel backend
+  if(n_cpu == -1) n_cpu <- detectCores() 
+  assert_that(detectCores() >= n_cpu)
+  
+  cl <- makeCluster(n_cpu)
```

Comment:
Nested parallel computation might not work. I would recommend putting n_cpu=1 for mmif in fit_and_score.

Reply:
done